app/src/main/java/org/wikipedia/talk/TalkPageSurvey.kt

```diff
@@ -0,0 +1,70 @@
+package org.wikipedia.talk
+
+import android.app.Activity
+import android.content.Context
+import android.widget.TextView
+import androidx.appcompat.app.AlertDialog
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.page.LinkMovementMethodExt
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.CustomTabsUtil
+import org.wikipedia.util.GeoUtil
+import org.wikipedia.util.StringUtil
+
+object TalkPageSurvey {
+
+    fun maybeShowSurvey(activity: Activity) {
+        if (!activity.isDestroyed && (fallsWithinGeoRange() || Prefs.talkPageSurveyOverride)) {
+            showSurveyDialog(activity)
+        }
+    }
+
+    fun showSurveyDialog(activity: Activity) {
+        val attempts = Prefs.showTalkPageSurveyAttempts
+        if (attempts > 1) {
+            return
+        }
+
+        Prefs.showTalkPageSurveyAttempts = attempts + 1
+
+        val dialog = AlertDialog.Builder(activity)
+                .setTitle(activity.getString(R.string.survey_dialog_title))
+                .setMessage(StringUtil.fromHtml(activity.getString(R.string.talk_snackbar_survey_text) +
+                        "<br/><br/><small><a href=\"https://foundation.m.wikimedia.org/wiki/Legal:Wikipedia_Android_App_Talk_Page_Survey_Privacy_Statement\">" +
+                        activity.getString(R.string.privacy_policy_description) + "</a></small>"))
+                .setPositiveButton(R.string.talk_snackbar_survey_action_text) { _, _ -> takeUserToSurvey(activity) }
+                .setNegativeButton(if (attempts == 0) R.string.onboarding_maybe_later else android.R.string.cancel, null)
+                .setCancelable(false)
+                .create()
+        dialog.findViewById<TextView>(android.R.id.message)?.movementMethod = LinkMovementMethodExt { url ->
+            CustomTabsUtil.openInCustomTab(activity, url)
+        }
+        dialog.show()
+    }
+
+    private fun fallsWithinGeoRange(): Boolean {
+        val languages = WikipediaApp.getInstance().language().appLanguageCodes
+        val country = GeoUtil.geoIPCountry.orEmpty()
+        return (languages.contains("hi") ||
+                languages.contains("id") ||
+                languages.contains("ja") ||
+                ((languages.contains("ar") || languages.contains("fr")) && (country == "MA" || country == "EG" || country == "ML" || country == "CD")) ||
+                (languages.contains("en") && (country == "IN" || country == "NG")))
+    }
+
+    private fun takeUserToSurvey(context: Context) {
```

Comment:
If takes the user to survey, should we add the value of the attempts to prevent seeing the dialog again?

Reply:
Another issue: 
first time of showing the dialog.
<img src="https://user-images.githubusercontent.com/2435576/139732343-b32335c8-c18b-4b10-9a25-5b538a44296f.jpg" width="400" />
second time of showing the dialog.
<img src="https://user-images.githubusercontent.com/2435576/139732346-723d9355-bdd7-4ceb-a1f1-772b505e6700.jpg" width="400" />

Reply:
The privacy policy link is not tappable.

Reply:
> If takes the user to survey, should we add the value of the attempts to prevent seeing the dialog again?

We do, don't we?

> The privacy policy link is not tappable.

Whoops, fixed that.

Not sure what you're showing with the screenshots. Is it the two buttons overflowing onto two rows? That is expected.

Reply:
> 
> 
> > If takes the user to survey, should we add the value of the attempts to prevent seeing the dialog again?
> 
> We do, don't we?
> 

Yes, we do. But if the user tap on the "Get started" for the first time and finishes the survey, they will still see the survey dialog because the value of the attempts is `1`.

> Not sure what you're showing with the screenshots. Is it the two buttons overflowing onto two rows? That is expected.

I was confused and it is not an issue for me now.

Reply:
> Yes, we do. But if the user tap on the "Get started" for the first time and finishes the survey, they will still see the survey dialog because the value of the attempts is `1`.

Ohhh of course, good catch. Fixed.